packages/wonder-blocks-dropdown/components/dropdown.test.js

```diff
@@ -0,0 +1,241 @@
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdownCore.instance().focusedIndex).toBe(2);
+    });
+
+    it("closes itself on escape", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+        const dropdownCore = dropdown.find(DropdownCore);
+
+        // Act
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.escape});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.escape});
+
+        // Assert
+        expect(dropdown.state("opened")).toBe(false);
+    });
+
+    it("closes itself on tab", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+        const dropdownCore = dropdown.find(DropdownCore);
+
+        // Act
+        dropdownCore.simulate("keydown", {keyCode: keyCodes.tab});
+        dropdownCore.simulate("keyup", {keyCode: keyCodes.tab});
+
+        // Assert
+        expect(dropdown.state("opened")).toBe(false);
+    });
+
+    it("closes itself on an external mouse click", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+
+        // Act
+        document.dispatchEvent(new MouseEvent("mouseup"));
+
+        // Assert
+        expect(dropdown.state("opened")).toBe(false);
+    });
+
+    it("open on down key when focused", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        const clickableBehavior = dropdown.find(ClickableBehavior);
+        clickableBehavior.simulate("click");
+
+        // Act
+        anchor.simulate("keydown", {keyCode: keyCodes.down});
+        anchor.simulate("keyup", {keyCode: keyCodes.down});
+
+        // Assert
+        expect(dropdown.state("opened")).toBe(true);
+    });
+
+    it.only("updates selectedValues when OptionItem clicked", () => {
+        // Arrange
+        const controlledComponent = mount(<ControlledComponent />);
+        controlledComponent.simulate("click");
+        dropdown = controlledComponent.find(DropdownCore);
+
+        // Act
+        dropdown.simulate("keydown", {keyCode: keyCodes.down});
+        dropdown.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(0);
+
+        const optionItem = controlledComponent.find(OptionItem).at(1);
+        optionItem.simulate("click");
+
+        // Assert
+        expect(controlledComponent.state("selectedValues").length).toBe(2);
+    });
+
+    it("removes selected values from selectedValues", () => {
+        // Arrange
+        const controlledComponent = mount(<ControlledComponent />);
+        controlledComponent.simulate("click");
+        dropdown = controlledComponent.find(DropdownCore);
+
+        // Act
+        dropdown.simulate("keydown", {keyCode: keyCodes.down});
+        dropdown.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(0);
```

Comment:
Keyboard navigation has already been tested in another test case so this assertion isn't necessary and can be removed.  Also, `Act` sections shouldn't have any assertions.  It's a sign that your test is trying to do too much.  Also, these "actions" should probably be part of the `Arrange` block b/c they're not the action we're trying to test as indicated by the title of the test.  I know some of our existing tests don't get this right all of the time